py/_io/terminalwriter.py

```diff
@@ -144,12 +144,30 @@ def __init__(self, file=None, stringio=False, encoding=None):
     def fullwidth(self):
         if hasattr(self, '_terminal_width'):
             return self._terminal_width
-        return get_terminal_width()
+        try:
+            width = self._cached_terminal_width
+        except AttributeError:
+            try:
+                signal.signal(signal.SIGWINCH, self._handle_sigwinch)
+            except AttributeError:  # SIGWINCH not supported.
+                width = self._cached_terminal_width = False
```

Comment:
after taking another looks i propose creating a global "cache" as per object basis will overwrite the signals of other terminal-writer objects and cause multiple objects not to get invalidated

in addition it may break interaction with other libraries as is since we don't call existing handlers